tests/ui/array_indexing.stderr

```diff
@@ -1,120 +1,222 @@
+error: indexing may panic. Consider using `.get(n)` or `.get_mut(n)` instead
+  --> $DIR/array_indexing.rs:11:5
+   |
+11 |     x[index];
+   |     ^^^^^^^^
+   |
+   = note: `-D indexing-slicing` implied by `-D warnings`
+
+error: slicing may panic. Consider using `.get(n..m)` or `.get_mut(n..m)` instead
+  --> $DIR/array_indexing.rs:12:6
+   |
+12 |     &x[index_from..index_to];
+   |      ^^^^^^^^^^^^^^^^^^^^^^^
+
+error: slicing may panic. Consider using `.get(..n)`or `.get_mut(..n)` instead
+  --> $DIR/array_indexing.rs:13:6
+   |
+13 |     &x[index_from..][..index_to];
+   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: slicing may panic. Consider using `.get(n..)` or .get_mut(n..)` instead
+  --> $DIR/array_indexing.rs:13:6
+   |
+13 |     &x[index_from..][..index_to];
+   |      ^^^^^^^^^^^^^^^
+
+error: slicing may panic. Consider using `.get(n..)` or .get_mut(n..)` instead
+  --> $DIR/array_indexing.rs:14:6
+   |
+14 |     &x[index..];
+   |      ^^^^^^^^^^
+
+error: slicing may panic. Consider using `.get(..n)`or `.get_mut(..n)` instead
+  --> $DIR/array_indexing.rs:15:6
+   |
+15 |     &x[..index];
+   |      ^^^^^^^^^^
+
 error: const index is out of bounds
-  --> $DIR/array_indexing.rs:12:5
+  --> $DIR/array_indexing.rs:18:5
    |
-12 |     x[4];
+18 |     x[4];
    |     ^^^^
    |
    = note: `-D out-of-bounds-indexing` implied by `-D warnings`
 
-error: const index is out of bounds
-  --> $DIR/array_indexing.rs:13:5
+error: range is out of bounds
+  --> $DIR/array_indexing.rs:20:6
    |
-13 |     x[1 << 3];
-   |     ^^^^^^^^^
+20 |     &x[1..5];
+   |      ^^^^^^^
 
-error: range is out of bounds
-  --> $DIR/array_indexing.rs:14:6
+error: slicing may panic. Consider using `.get(n..m)` or `.get_mut(n..m)` instead
+  --> $DIR/array_indexing.rs:20:6
    |
-14 |     &x[1..5];
+20 |     &x[1..5];
    |      ^^^^^^^
 
-error: range is out of bounds
-  --> $DIR/array_indexing.rs:16:6
+error: slicing may panic. Consider using `.get(..n)`or `.get_mut(..n)` instead
+  --> $DIR/array_indexing.rs:21:6
    |
-16 |     &x[0..=4];
-   |      ^^^^^^^^
+21 |     &x[1..][..5];
+   |      ^^^^^^^^^^^
+
+error: slicing may panic. Consider using `.get(..n)`or `.get_mut(..n)` instead
+  --> $DIR/array_indexing.rs:23:6
+   |
+23 |     &x[0..][..3];
+   |      ^^^^^^^^^^^
```

Comment:
What happens if you add `&x[0..].get(..3)` as another test case. Does the lint get triggered on `x[0..]`?